File: flyweight/python/box.py
from abc import ABC, abstractmethod
from box_color import BoxColor


class Box(ABC):
    """
    Nesnenin ortak özelliklerini tutan sınıftır.
    UML diyagramındaki Flyweight yapısına denk gelmektedir.
    """
    width: int
    height: int
    box_color: BoxColor

    @abstractmethod
    def draw(self, location_x, location_y):
        pass

File: flyweight/typescript/box.ts
import { BoxColor } from "./box-color";

// Nesnenin ortak özelliklerini tutan sınıftır.
// UML diyagramındaki Flyweight yapısına denk gelmektedir.
export abstract class Box {
  public width: number;
  public height: number;
  public boxColor: BoxColor;

  public abstract draw(locationX: number, locationY: number): void;
}
